File: lesson_3/code/update_position.py
import random


def update_position(x: int, y: int, area: int) -> tuple[int, int]:
    direction = random.randint(1, 4)
    if direction == 1:
        if y < area:
            y += 1
    elif direction == 2:
        if y > 0:
            y -= 1
    elif direction == 3:
        if x > 0:
            x -= 1
    else:
        if x < area:
            x += 1
    return x, y

File: lesson_3/code/update_position.m
function [x, y] = update_position(x, y, area)
    
direction = randi([1 4]);  
    switch direction    
        case 1  % go up
            if y < area; y = y + 1; 
            end

        case 2  % go down
            if y > 0; y = y - 1;
            end
            
        case 3  % go left
            if x > 0; x = x - 1;
            end
             
        case 4  % go right
            if x < area; x = x + 1;
            end
    end
end